tests/integration/targets/rabbitmq_upgrade/tasks/tests.yml

```
@@ -0,0 +1,91 @@
+- block:
+  - set_fact:
+      parameter_node: rabbit
+
+  - name: Drain node
+    rabbitmq_upgrade:
+      action: drain
+      node: "{{ parameter_node }}"
+    register: result
+
+  - name: Check if node was properly drained
+    assert:
+      that:
+        - result is success
+        - result is changed
+
+  - name: Ensure node is under maintenance
+    shell: "rabbitmq-diagnostics -n {{ parameter_node }} status | grep 'Is under maintenance?: true'"
+
+  - name: Drain node (idempotency)
+    rabbitmq_upgrade:
+      action: drain
+      node: "{{ parameter_node }}"
+    register: result
+
+  - name: Check idempotency
+    assert:
+      that:
+        - result is not changed
+
+  - name: Revive node
+    rabbitmq_upgrade:
+      action: revive
+      node: "{{ parameter_node }}"
+    register: result
+
+  - name: Check if node was properly revived
+    assert:
+      that:
+        - result is success
+        - result is changed
+
+  - name: Ensure node is under maintenance
+    shell: "rabbitmq-diagnostics -n {{ parameter_node }} status | grep 'Is under maintenance?: false'"
+
+  - name: Revive node (idempotency)
+    rabbitmq_upgrade:
+      action: revive
+      node: "{{ parameter_node }}"
+    register: result
+
+  - name: Check idempotency
+    assert:
+      that:
+        - result is not changed
+
+  - name: Execute await_online_quorum_plus_one
+    rabbitmq_upgrade:
+      action: await_online_quorum_plus_one
+      node: "{{ parameter_node }}"
+    register: result
```

```suggestion
    register: result
```
Would be nice to cover check mode

---

but unfortunately, i have no idea how to do it, do we have some example?

---

just create similar tasks but with `check_mode: yes`

---

Hey, I've covered check_mode in integration tests

But could You please explain my why are we doing this? I'm trying to find any particular case when check_mode can return different 'changed' value?
But I'm not super familiar with ansible(especially when it comes to testing/writing modules) so maybe I don't understand something.

---

Yes, sure, I can. We should cover all possible cases (i.e. options and their combinations + check_mode + even fail messages). If a module supports check_mode, it should be covered. This is a kind of general idea of testing. If once someone decides to refactor your module, changes related code, and it will break the check_mode anyhow, we will know about it.
In case of this module, the check_mode implementation is pretty straightforward but anyway we should be sure that module returns in check_mode what we're expecting and protect it from further breaking changes.

---

it's also reflected somewhere in the doc

---

ok, thanks for the explanation!

---

You're welcome!